packages/@lwc/engine/src/framework/vm.ts

```diff
@@ -162,6 +167,11 @@ function resetComponentStateWhenRemoved(vm: VM) {
     }
     const { state } = vm;
     if (state !== VMState.disconnected) {
+        const { tro } = vm;
+        // Making sure that any observing record will not trigger the rehydrated on this vm
+        if (!isNull(tro)) {
```

Comment:
in which case this function is called the template reactive observer being `null`?

Reply:
good call, it should never be null